eclair-core/src/main/scala/fr/acinq/eclair/router/Graph.scala

```diff
@@ -625,27 +808,14 @@ object Graph {
        *
        * @param channels map of all known public channels in the network.
        */
-      def makeGraph(channels: SortedMap[RealShortChannelId, PublicChannel]): DirectedGraph = {
-        // initialize the map with the appropriate size to avoid resizing during the graph initialization
-        val mutableMap = new mutable.HashMap[PublicKey, List[GraphEdge]](initialCapacity = channels.size + 1, mutable.HashMap.defaultLoadFactor)
-
-        // add all the vertices and edges in one go
-        channels.values.foreach { channel =>
-          channel.update_1_opt.foreach(u1 => addToMap(GraphEdge(u1, channel)))
-          channel.update_2_opt.foreach(u2 => addToMap(GraphEdge(u2, channel)))
-        }
-
-        def addToMap(edge: GraphEdge): Unit = {
-          mutableMap.put(edge.desc.b, edge +: mutableMap.getOrElse(edge.desc.b, List.empty[GraphEdge]))
-          if (!mutableMap.contains(edge.desc.a)) {
-            mutableMap += edge.desc.a -> List.empty[GraphEdge]
-          }
-        }
+      def makeGraph(channels: Map[RealShortChannelId, PublicChannel], nodes: Map[PublicKey, NodeAnnouncement]): DirectedGraph = {
+        val edges = channels.values.flatMap(channel =>
+          Seq(channel.update_1_opt.map(ActiveEdge(_, channel)), channel.update_2_opt.map(ActiveEdge(_, channel))).flatten)
```

Comment:
We potentially have channel updates in the DB that are disabled, you need to avoid creating active edges for those (a unit test would be desirable here):

```suggestion
        val edges = channels.values.flatMap(channel => Seq(
          channel.update_1_opt.collect { case u1 if u1.channelFlags.isEnabled => ActiveEdge(u1, channel) },
          channel.update_2_opt.collect { case u2 if u2.channelFlags.isEnabled => ActiveEdge(u2, channel) },
        ).flatten)
```

Reply:
Fixed, and added a unit test.